Makefile

```diff
@@ -0,0 +1,101 @@
+# Copyright 2021 Gravitational, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+MAKEFILE := $(abspath $(lastword $(MAKEFILE_LIST)))
+ROOTDIR := $(patsubst %/,%,$(dir $(MAKEFILE)))
+
+ifeq ($(origin VERSION), undefined)
+# avoid lazily evaluation (and thus rerunning the shell command several times)
+VERSION := $(shell ./version.sh)
+endif
+
+ifeq ($(origin COMMIT), undefined)
+COMMIT := $(shell git rev-parse HEAD)
+endif
+
+BUILDDIR := $(ROOTDIR)/build
+OUT := $(BUILDDIR)/drone-fork-approval-extension
+GOSRC := $(shell find $(ROOTDIR) -type f -name '*.go') go.mod go.sum
+LDFLAGS=-ldflags "-X=main.version=$(VERSION) -X=main.commit=$(COMMIT)"
```

Comment:
Curious: are the `-X` parameters intended to help with reproducible builds?

Reply:
> Curious: are the `-X` parameters intended to help with reproducible builds?

No. Using ldflags this was is worse for reproducibility, because they inject dynamic values into otherwise static source code.  In this case the commit doesn't matter (since changing the source code would also change the commit hash), but the tag is derived from git describe, which depends on info outside git's commit/tree.  That means rebuilding a commit could result in a different build artifact depending on the presence or absence of a tag in that commit's history.  Not great. :(

In practice, this won't matter so as long as the following holds true:
* Tags aren't rewritten
* Tags aren't added for old commits
* Untagged commits aren't published